### ISSUE TITLE
Pull assumption/assert generation out of astconv.

### ISSUE THREAD

**Issue by UnsignedByte** [cucapra/filament#264]
Discussed in slack; astconv currently generates assumptions and asserts for things like let parameters, but we might want to generate assumptions and asserts post-monomorphization, which would require us to move these assumptions out of astconv into its own pass.

**Comment by rachitnigam** [comment]
Yeah! This is also important to get parity with the original flow which ran type checking before and after monomorphization (as a sanity check).
